packages/gcf-utils/src/bot-request.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as express from 'express';
+import {SCHEDULER_EVENT_NAMES} from './custom-events';
+
+/**
+ * Type of function execution trigger
+ */
+export enum TriggerType {
+  GITHUB = 'GitHub Webhook',
+  SCHEDULER = 'Cloud Scheduler',
+  TASK = 'Cloud Task',
+  PUBSUB = 'Pub/Sub',
+  UNKNOWN = 'Unknown',
+}
+
+export interface BotRequest {
+  eventName: string;
+  githubDeliveryId: string;
+  signature?: string;
+  taskName?: string;
+  taskRetryCount?: number;
+  traceId?: string;
+  triggerType: TriggerType;
+}
+
+export function parseBotRequest(request: express.Request): BotRequest {
+  const eventName =
+    request.get('x-github-event') || request.get('X-GitHub-Event') || '';
```

Comment:
Test, but this should be case-insensitive:

> Returns the specified HTTP request header field (case-insensitive match). The Referrer and Referer fields are interchangeable.

https://expressjs.com/en/5x/api.html